x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -603,6 +610,7 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
                         <RuleSwitch
                           id={rule?.id ?? '-1'}
                           isDisabled={
+                            !rule ||
```

Comment:
Global loader is considered as bad UX nowadays and skeleton loading is much more preferable. We display as much as we can as early as possible to avoid the loader flickering. If you still think it's worth to show some loader instead I suggest to discuss it out of scope of this PR.